projects/dotcms-js/src/lib/core/core-web.service.ts

```diff
@@ -48,40 +50,52 @@ export const RULE_CONDITION_UPDATE_TYPE = 'RULE_CONDITION_UPDATE_TYPE';
 export const RULE_CONDITION_UPDATE_PARAMETER = 'RULE_CONDITION_UPDATE_PARAMETER';
 export const RULE_CONDITION_UPDATE_OPERATOR = 'RULE_CONDITION_UPDATE_OPERATOR';
 
+export interface RequestOptionsParams {
+    headers?: HttpHeaders;
+    reportProgress?: boolean;
+    params?: HttpParams;
+    responseType?: 'arraybuffer' | 'blob' | 'text' | 'json';
+    withCredentials?: boolean;
+}
+
 @Injectable()
 export class CoreWebService {
     private httpErrosSubjects: Subject<any>[] = [];
 
     constructor(
         private _apiRoot: ApiRoot,
-        private _http: Http,
         private loggerService: LoggerService,
         private browserUtil: BrowserUtil,
-        private router: Router
+        private router: Router,
+        private http: HttpClient
     ) {}
 
-    request(options: any): Observable<any> {
+    request<F>(options: RequestOptionsArgs): Observable<F | any> {
         const request = this.getRequestOpts(options);
         const source = options.body;
 
-        return this._http.request(request).pipe(
-            map((resp: Response) => {
+        return this.http.request(request).pipe(
+            filter(<T>(event: HttpEvent<T>) => event.type === HttpEventType.Response),
```

Comment:
Sorry again on this but, I don't see this `filter` in the angular docs: https://angular.io/guide/http where did you saw that approach?